DashKit/DashPumpManager.swift

```diff
@@ -680,26 +684,19 @@ open class DashPumpManager: PumpManager {
                 case .success:
                     self.log.default("Recommending Loop")
                     self.finalizeAndStoreDoses()
-                    self.pumpDelegate.notify({ (delegate) in
-                        completion?()
-                        delegate?.pumpManagerRecommendsLoop(self)
-                    })
+                    completion?(self.lastSync)
                 case .failure(let error):
                     self.log.default("Not recommending Loop because pump data is stale: %@", String(describing: error))
+                    completion?(self.lastSync)
```

Comment:
[nit] should this `completion?(self.lastSync)` call be inside the `notify` block below?

Reply:
The completion handler doesn't have any specific constraints on what thread it calls on; the notify block is a mechanism for calling methods on the delegate.